### PR TITLE
[Swift bindings] Make projection tooling AOT compatible

### PR DESCRIPTION
**Pull request by kotlarmilos**
## Description

This PR sets the `IsAotCompatible` property and updates the code to be trim-compatible. This ensures that future changes will implement trim-compatible code.

### CODE REVIEW

File: src/Swift.Bindings/src/Marshaler/Conductor.cs [hunk 1]
```diff
@@ -17,22 +17,24 @@ namespace BindingsGeneration
     /// It initializes the handler factories and provides methods to fetch handlers for given declarations.
     /// </summary>
     public class Conductor {
-        private readonly List<IModuleHandlerFactory> _moduleHandlerFactories = [];
-        private readonly List<ITypeHandlerFactory> _typeHandlerFactories = [];
+        private readonly List<IModuleHandlerFactory> _moduleHandlerFactories = [
```
Comment (jkurdek) [inline src/Swift.Bindings/src/Marshaler/Conductor.cs:20]: Im not sure those were necessary. Do we also require the tooling to be trim-friendly / native-aot friendly? I thought that the NativeAOT support requirement is only specified on the output bindings / Swift.Runtime code.

But on the other hand I think it simplifies the design and takes some magic away.

Comment (kotlarmilos) [reply src/Swift.Bindings/src/Marshaler/Conductor.cs:20]: Not necessary, but nice to have. It is easy to set requirements now as the codebase is still small. @stephen-hawley What do you think?

Comment (kotlarmilos) [reply src/Swift.Bindings/src/Marshaler/Conductor.cs:20]: Added AOT compatible property to tests (https://github.com/dotnet/runtimelab/pull/2871/commits/dcf3fdb7c75cf7b44f67463dd28dacb0bc448265) and experimental NuGet (https://github.com/dotnet/runtimelab/pull/2878/commits/ae322434816a84a4121071603c0ced47d47e9604).

Comment (stephen-hawley) [reply src/Swift.Bindings/src/Marshaler/Conductor.cs:20]: I don't see it as a requirement for the tooling, no.

Comment (jkotas) [reply src/Swift.Bindings/src/Marshaler/Conductor.cs:20]: I agree that it is not a strict requirement, but it is very nice to have. Native AOT makes the tools like compilers a lot faster. We compile number of shipping tools (e.g. the AOT compiler itself) using native AOT for this reason.

**Review by jkurdek — APPROVED** [review]

**Review by stephen-hawley — APPROVED** [review]

**Pull request merged**